lib/features/support/diagnostics.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'json'
+
+After do |scenario|
+  if scenario.failed?
+    if Server.stored_requests.empty?
+      $logger.info 'No requests received'
+    else
+      $logger.info 'The following requests were received:'
+      Server.stored_requests.each_with_index do |request, number|
+        json = JSON.pretty_generate request
+        $logger.info "Request #{number}: \n#{json}"
```

Comment:
It doesn't really matter, but it might be nice for this to start at "Request 1" rather than "Request 0"

```suggestion
      Server.stored_requests.each.with_index(1) do |request, number|
        json = JSON.pretty_generate request
        $logger.info "Request #{number}: \n#{json}"
```